Revisions to README and directives

awholegunch opened the pull request:
Major revision and update to the README and directives.

Once this is approved, there will be a companion one to update the Java files and their annotations.

Commit:
w-i-p

Commit:
Update README.md

fix links

Commit:
Update README.md

Commit:
Remove snapshot

Commit:
changes per comments.

Commit:
Added missing copyrights statements.
Edits per comments.
Added a Slack Team link.

Commit:
Add missing or adjust formatting (remove extra leading double spaces)
of copyright statements.

Commit:
Edits (forgot to push)

Commit:
Updates

Commit:
Merge branch 'release/1.3' into docs/revisions

Commit:
Edits

Commit:
Merge pull request #77 from hydrator/docs/revisions

Revisions

Commit:
w-i-p

Commit:
w-i-p

Commit:
Merge branch 'release/1.3' into docs/revisions_directives

# Conflicts:
#	README.md (used local version)

Commit:
Fix table.

Commit:
w-i-p

Commit:
re-write

nitinmotgi commented:
why directives are not upper case ? There is a reason why I had them the way I had. Could you please explain your reason and also did u understand why I had them the way I had them @awholegunch

nitinmotgi commented:
Also merge it to develop

russorat commented on docs/directives/cleanse-column-names.md:
```diff
@@ -1,49 +1,43 @@
-  }
+{
+  "col1": 1,
+  "col_2": 2
```
need commas at the end of these lines

awholegunch replied:
Fixed.

russorat commented on docs/directives/cleanse-column-names.md:
```diff
@@ -1,49 +1,43 @@
-  }
+{
+  "COL1": 1,
+  "col:2": 2
```
commas

awholegunch replied:
Fixed.

russorat commented on docs/directives/drop.md:
````diff
@@ -1,44 +1,44 @@
 
-```column``` is the name of the column in the record to be droped. If the ```column``` does not exist in the record, then the directive fails.
+The `<column>` is the name of the column in the record to be droped. If the `<column>`
+does not exist in the record, the directive fails.
````
technically, if you drop a column that doesn't exist, it just doesn't do anything and continues processing. It doesn't actually give an error message.

awholegunch replied:
Fixed.

russorat commented on docs/directives/filter-rows-on.md:
````diff
@@ -1,56 +1,57 @@
+Applying this directive:
 ```
-  filter-row-if-true country !~ US
+filter-rows-on condition-true country !~ US
````
i think you need single quotes around US

awholegunch replied:
Fixed. Plus a few other errors on that page.

russorat commented on docs/directives/hash.md:
```diff
@@ -1,105 +1,105 @@
+* KECCAK-384
 * KECCAK-512
+* MD2
+* MD2
```
are these repeated? same with MD5

awholegunch replied:
Fixed.

russorat commented on docs/directives/hash.md:
```diff
@@ -1,105 +1,105 @@
 * SHA-1
+* SHA-224
+* SHA-256
+* SHA-256
```
also duplicated

awholegunch replied:
Fixed (and others!)

russorat commented on docs/directives/merge.md:
```diff
@@ -1,56 +1,52 @@
-    "b" : 2
-  }
+  "first": "Root",
+  "last": "Joltie"
```
commas

awholegunch replied:
Fixed.

russorat commented on docs/directives/parse-as-csv.md:
````diff
@@ -0,0 +1,52 @@
+
+## Syntax
+```
+parse-as-csv <column-name> <delimiter> <skip-on-error>
````
I think skip-on-error is actually parse headers now. parse-as-csv <column> <delimiter> [<header=true/false>]

russorat commented on docs/directives/parse-as-date.md:
```diff
@@ -1,27 +1,47 @@
+It recognizes dates described in many ways, including formal dates (`02/28/1979`), relaxed dates (`oct 1st`),
+relative dates (`the day before next thursday`), and even date alternatives (`next wed or thurs`).
 
 This directive will also search for date components within a larger block of text, detect their structure, and
```
Not sure if relevant, but it might be good to note that you need to convert this back to a string before adding to a pipeline.

awholegunch replied:
Added.

awholegunch commented:
@nitinmotgi "why directives are not upper case ? There is a reason why I had them the way I had. Could you please explain your reason and also did u understand why I had them the way I had them"

I made them the same case as they appear in the UI. I did it to be consistent. What was you reason for having them different?

awholegunch commented:
The other change I made was to make them all literals, so that they would stand out when rendered in a browser.

Plus, lowercase is easier to read than all-caps, as then you have ascenders and descenders:
```
find-and-replace versus FIND-AND-REPLACE

Commit:
Added missing item to README file.
Added missing commas in JSON examples.
Updated drop to remove error message.
Removed duplicate entries in hash.
Adjusted parse-as-csv to correct the syntax, changed example accordingly
and added another example.

Commit:
Correct items that say they have an error.

awholegunch commented:
@russorat @nitinmotgi I've updated the files based on the comments, and corrected additional errors that I found. Please take another look.

Commit:
Corrected examples, adjusted order of filters, and fixed error in
directive.

Commit:
Add note about usage of filter-rows-on

awholegunch commented:
@nitinmotgi Also, you asked that this be merged to `develop` rather than `release/1.3`; is there a reason for that? Does it not make sense to get these out sooner?

nitinmotgi commented:
@awholegunch I am applying all changes to develop as there are serious changes in API. So, it will be moving to release 2.0 of plugin.

awholegunch commented:
@nitinmotgi Got it.

nitinmotgi commented:
In regards to directive names, I wanted to stand them out that's the reason. It's a general practice I have seen around the web.

Commit:
Change to cut-character

Commit:
Merge branch 'develop' into docs/revisions_directives

# Conflicts:
#	README.md
#	core/pom.xml
#	core/src/main/java/co/cask/wrangler/steps/language/SetCharset.java
#	core/src/main/java/co/cask/wrangler/steps/row/SetRecordDelimiter.java
#	docs/directives/find-and-replace.md
#	docs/directives/mask-number.md
#	docs/directives/merge.md
#	docs/directives/stemming.md
#	pom.xml
#	service/pom.xml
#	transform/pom.xml

Commit:
Use capitalization in directives. Other edits and fixes.

Commit:
Merge branch 'develop' into docs/revisions_directives

# Conflicts:
#	README.md
#   resolved using mine.

Commit:
Edits.

awholegunch commented:
This has been updated with capitalizations. There should be no additional changes to this PR except in response to comments; I have started a new branch for Java code changes.

Commit:
Update split-to-columns.md

Add link and match usage of Java code.

Commit:
Update swap.md

Matches usage.

nitinmotgi reviewed (approved):
lgtm
